src/Symfony/Component/ClassLoader/ClassCollectionLoader.php

```diff
@@ -90,12 +90,37 @@ public static function load($classes, $cacheDir, $name, $autoReload, $adaptive =
             return;
         }
 
+        self::inline($classes, $cache, $declared);
+
+        if ($autoReload) {
+            // save the resources
+            self::writeCacheFile($metadata, serialize(array($files, $classes)));
```

Comment:
Good catch, fixed by making inline() return $files. The cache invalidation logic should remain in load() so I kept the general logic as is.